src/libcore/iter.rs

```diff
@@ -280,6 +289,16 @@ pub trait Iterator {
     /// different sized integer, the `zip` function provides similar
     /// functionality.
     ///
+    /// # Undefined overflow
+    ///
+    /// The method does no guarding against overflows, so enumerating more than
+    /// `usize::MAX` elements is undefined.
```

Comment:
@huonw I can't find the intention for overflow to be wrapping in the RFC thread (other than as a consequence of how `rustc` implements it currently). We can e.g. see this user suggesting what you said, but nobody telling that this is indeed the RFC's text (so it seems it isn't the intention):

https://github.com/rust-lang/rfcs/pull/560#issuecomment-69446912:
>I like this proposal, but I don't like the concept of "unspecified behavior" that isn't really unspecified. To avoid confusion, I'd prefer to say that if the overflow is not diagnosed, the value is guaranteed to wrap.
>Alternatively, you could write that it truly is unspecified, and optimize based on that, but I think that's non-ideal.

Now, reading between the lines of the following, it seems that the following assumes that `rustc`'s implementation will be wrap-on-overflow, but that this isn't meant to be *guaranteed*.

https://internals.rust-lang.org/t/a-tale-of-twos-complement/1062
>Aren't you concerned about de facto lock-in for overflow?
>
>There is certainly a concern that we will not be able to make overflow stricter without breaking code in the wild. We believe that having overflow checking be mandatory in debug builds should at least mitigate that risk, as intentional uses of overflow should be detected long before the code ever comes into widespread use (and redirected to the suitable wrapper type).

The RFC as it stands right now doesn't sound like "wrap or panic" semantics, but rather than "panic or unspecified value" which is a superset of the former, so we don't lock ourselves regarding backward compatiblity if we don't guarantee this right now.